app/controllers/article_controller.rb

```diff
@@ -91,4 +83,18 @@ def search_service
   def set_search_query_modifier
     @search_modifier ||= SearchQueryModifier.new(params, blacklight_config)
   end
+
+  # Reuse the EDS session token if available in the user's session data,
+  # otherwise establish a session
+  def setup_eds_session(session)
+    return if session['eds_session_token'].present?
+    session['eds_session_token'] = EBSCO::EDS::Session.new(
+      guest: session['guest'],
```

Comment:
There's also some callback in `devise-guests` when a guest user logs in you might use to update the session. `logging_in_user`, maybe?